File: python/Due_/DistanceSensor.py

class DistanceSensorController:
    def __init__(self, serialPort):
        self.serialPort = serialPort
        self.MAX_IO = 20

    def Read(self, pulsePin, echoPin):

        if pulsePin < 0 or pulsePin >= self.MAX_IO:
            raise ValueError('Invalid pin')

        if echoPin < 0 or echoPin >= self.MAX_IO:
            raise ValueError('Invalid pin')

        cmd = f'print(distance({pulsePin},{echoPin}))'
        self.serialPort.WriteCommand(cmd)

        res = self.serialPort.ReadRespone()

        if res.success:
            try:
                distance = int(res.respone)
                return distance
            except ValueError:
                pass

        return -1

File: dotnet/GHIElectronics.DUE/DistanceSensor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static GHIElectronics.DUE.DUEController;

namespace GHIElectronics.DUE {
    public partial class DUEController {
        public class DistanceSensorController {


            SerialInterface serialPort;
            public DistanceSensorController(SerialInterface serialPort) => this.serialPort = serialPort;

            public int Read(int pulsePin, int echoPin) {
                if (pulsePin < 0 || pulsePin >= this.serialPort.DeviceConfig.MaxPinIO)
                    throw new ArgumentOutOfRangeException("Invalid pin.");

                if (echoPin < 0 || echoPin >= this.serialPort.DeviceConfig.MaxPinIO)
                    throw new ArgumentOutOfRangeException("Invalid pin.");

                var cmd = string.Format("print(distance({0},{1}))", pulsePin, echoPin);

                this.serialPort.WriteCommand(cmd);

                var res = this.serialPort.ReadRespone();

                if (res.success) {
                    try {
                        var distance = int.Parse(res.respone);
                        return distance;
                    }
                    catch {

                    }

                }

                return -1;
            }
        }
    }
}
